### PR TITLE
Require Elixir v1.4 + rely on the new application inference

### PR DESCRIPTION
**Pull request by jamilabreu**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -6,7 +6,7 @@ defmodule Phoenix.Mixfile do
   def project do
     [app: :phoenix,
      version: @version,
-     elixir: "~> 1.3",
+     elixir: "~> 1.4",
```
Comment (josevalim) [inline mix.exs]: Let's keep the version here `~> 1.3` so we keep people upgrading a more breathing room but people starting a new app start fresh.

File: mix.exs [hunk 2]
```diff
@@ -34,7 +34,7 @@ defmodule Phoenix.Mixfile do
 
   def application do
     [mod: {Phoenix, []},
-     applications: [:plug, :poison, :logger, :eex],
+     extra_applications: [:logger, :eex],
```
Comment (josevalim) [inline mix.exs]: Therefore, let's revert this change BUT leave a TODO so we update it as soon as 1.3 is out.

**Comment by josevalim** [comment]
Thank you! I have added two comments! If you fix them you will likely fix travis too. :)

**Comment by jamilabreu** [comment]
✅

**Comment by jamilabreu** [comment]
One check passing so far - others caused I believe by mismatch in versions

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
